### PR TITLE
Improve detection of the target architecture on Apple platforms.

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
This PR moves checking the `CMAKE_OSX_ARCHITECTURES` variable before checking the path of the C compiler. Likely fixes detecting the correct architecture when cross-compiling to arm64 from a x64 machine.

### CODE REVIEW

File: scripts/buildsystems/vcpkg.cmake [hunk 1]
```diff
@@ -268,6 +268,39 @@ else()
         set(Z_VCPKG_TARGET_TRIPLET_ARCH arm64)
     elseif(CMAKE_GENERATOR STREQUAL "Visual Studio 17 2022")
         set(Z_VCPKG_TARGET_TRIPLET_ARCH x64)
+    elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL "Darwin" AND APPLE)
+        list(LENGTH CMAKE_OSX_ARCHITECTURES Z_VCPKG_OSX_ARCH_COUNT)
```
[the remark this answers is not in the document]
Comment (teo-tsirpanis) [reply scripts/buildsystems/vcpkg.cmake:272]: Done.

File: scripts/buildsystems/vcpkg.cmake [hunk 2]
```diff
@@ -278,39 +311,6 @@ else()
             set(Z_VCPKG_TARGET_TRIPLET_ARCH arm64)
         elseif(Z_VCPKG_CL MATCHES "bin/cl.exe$" OR Z_VCPKG_CL MATCHES "x86/cl.exe$")
             set(Z_VCPKG_TARGET_TRIPLET_ARCH x86)
-        elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL "Darwin" AND DEFINED CMAKE_SYSTEM_NAME AND NOT CMAKE_SYSTEM_NAME STREQUAL "Darwin")
```
[the remark this answers is not in the document]
Comment (teo-tsirpanis) [reply scripts/buildsystems/vcpkg.cmake:281]: > this was meant to match non-macOS targets only

I can't see a reason for it, `CMAKE_OSX_ARCHITECTURES` applies to all kinds of Apple platforms.

**Review by dg0yt — APPROVED** [review]
LGTM but someone else must approve.

**Comment by BillyONeal** [comment]
Thanks for the fix!

**Pull request merged**